packages/core/src/fhirpath/functions.ts

```diff
@@ -1791,6 +1791,81 @@ export const functions: Record<string, FhirPathFunction> = {
       value: value.value?.resourceType === expectedResourceType,
     }));
   },
+
+  /*
+   * SQL-on-FHIR utilities
+   */
+
+  /**
+   * Returns an opaque value to be used as the primary key for the row associated with the resource.
+   *
+   * In many cases the value may just be the resource id, but exceptions are described below.
+   * This function is used in tandem with getReferenceKey(), which returns an equal value from references that point to this resource.
+   *
+   * The returned KeyType is implementation dependent, but must be a FHIR primitive type that can be used for efficient joins in the
+   * system’s underlying data storage. Integers, strings, UUIDs, and other primitive types are appropriate.
+   *
+   * See: https://build.fhir.org/ig/FHIR/sql-on-fhir-v2/StructureDefinition-ViewDefinition.html#getresourcekey--keytype
+   *
+   * @param _context - The evaluation context.
+   * @param input - The input collection.
+   * @returns The resource key.
+   */
+  getResourceKey: (_context: AtomContext, input: TypedValue[]): TypedValue[] => {
+    const resource = input[0].value as Resource;
+    if (!resource?.id) {
+      return [];
+    }
+    return [{ type: PropertyType.id, value: resource.id }];
```

Comment:
Do these need to be globally unique, or just per resource type?

Reply:
Good question.  Using `id` is the recommended implementation pattern from the spec: https://build.fhir.org/ig/FHIR/sql-on-fhir-v2/StructureDefinition-ViewDefinition.html#getresourcekey--keytype

The spec does include language about how that could lead to ambiguity if ID's are not globally unique.

I'm kinda mixed on the pros/cons.  Given that the spec recommended using `id`, I'm inclined to go with it 🤷

Reply:
Cool, I'm alright with using just `id` — especially since this might be less of a risk if we switch to V5 UUIDs in the future for more "guaranteed" global uniqueness